contracts/StreamManager.sol

```diff
@@ -38,12 +47,38 @@ contract StreamManager is IStreamManager {
             _payee,
             _token,
             _rate,
-            _terminationPeriod
+            _terminationPeriod,
+            true
         );
         address streamInstance = address(openStreamInstance);
+        /// @dev stores the address of the payee, and the address of his flow instance
+        streams[_payee] = streamInstance;
         /// @dev when creating an instance, it deposits stable coins(USDC or USDT)
         IERC20(_token).safeTransferFrom(msg.sender, streamInstance, _amount);
 
         emit OpenStreamCreated(msg.sender, streamInstance);
     }
+
+    /**
+     * @dev Payer can cancel open stream instance
+     * @param _payee payee address
+     */
+    function cancelOpenStream(address _payee) external {
+        /// @dev address of the open stream instance
+        address streamInstance = streams[_payee];
+        /// @dev getting the balance of the token USDC or USDT
+        uint256 amount = IOpenStream(streamInstance).getTokenBanance();
+
+        require(_payee != address(0), "Stream Manager: invalid address");
```

Comment:
Starting from Solidity v0.8.4, there is a convenient and gas-efficient
way to describe to users why an operation failed through the use of
custom errors. The require statement uses strings to provide additional
information about failures (e.g. require(blockings[user] == true, "non
blocked user");), but they are rather expensive, especially when it comes
to deploying cost, and it is difficult to use dynamic information in them.
BVSS:
AO:S/AC:L/AX:L/C:N/I:N/A:N/D:N/Y:N/R:N/S:U (0.0)
Recommendation:
It is recommended to implement custom errors instead of reverting strings.
References:
https://blog.soliditylang.org/2021/04/21/custom-errors/